medusa/name_parser/parser.py

```diff
@@ -156,53 +175,59 @@ def _parse_string(self, name):
+                new_episode_numbers.extend(episode)
+                new_season_numbers.append(season)
 
         elif result.season_number and result.episode_numbers:
             for episode_number in result.episode_numbers:
-                s = result.season_number
-                e = episode_number
+                season = result.season_number
+                episode = episode_number
 
                 if result.show.is_scene:
-                    (s, e) = scene_numbering.get_indexer_numbering(result.show.indexerid,
-                                                                   result.show.indexer,
-                                                                   result.season_number,
-                                                                   episode_number)
-                    logger.debug("Scene numbering enabled show '{name}' using indexer numbering: {ep}",
-                                 name=result.show.name, ep=episode_num(s, e))
+                    (season, episode) = scene_numbering.get_indexer_numbering(
+                        result.show.indexerid,
+                        result.show.indexer,
+                        result.season_number,
+                        episode_number
+                    )
+                    log.debug(
+                        'Scene numbering enabled series {name} using indexer numbering: {ep}',
+                        {'name': result.show.name, 'ep': episode_num(season, episode)}
+                    )
 
                 if result.show.is_anime:
-                    a = helpers.get_absolute_number_from_season_and_episode(result.show, s, e)
+                    a = helpers.get_absolute_number_from_season_and_episode(result.show, season, episode)
                     if a:
                         new_absolute_numbers.append(a)
-                        logger.debug("Scene numbering enabled anime show '{name}' using indexer with absolute {absolute}: {ep}",
-                                     name=result.show.name, absolute=a, ep=episode_num(s, e, 'absolute'))
+                        log.debug(
+                            'Scene numbering enabled anime {name} using indexer with absolute {absolute}: {ep}',
+                            {'name': result.show.name, 'absolute': a, 'ep': episode_num(season, episode, 'absolute')}
+                        )
 
-                new_episode_numbers.append(e)
-                new_season_numbers.append(s)
+                new_episode_numbers.append(episode)
+                new_season_numbers.append(season)
 
         # need to do a quick sanity check heregex.  It's possible that we now have episodes
         # from more than one season (by tvdb numbering), and this is just too much
         # for the application, so we'd need to flag it.
-        new_season_numbers = list(set(new_season_numbers))  # remove duplicates
+        new_season_numbers = sorted(set(new_season_numbers))  # remove duplicates
```

Comment:
Okay I checked this. Was concerned at first but it returns a list. So seems safe.